app/utils/get_filtered_records.py

```diff
@@ -246,11 +246,13 @@ def grab_cols(result, columns):
 '''
 
 
-def get_paginated_records(query_dicts, sorting_key, page_index, per_page, reverse):
+def get_paginated_records(query_dicts, sorting_key, min_page_index, max_page_index, per_page, reverse):
     # Order the records first
     sorted_records = sorted(query_dicts, key=lambda x: (x[sorting_key] is None, x[sorting_key]), reverse=reverse)
 
-    start = page_index * per_page
-    end = page_index * per_page + per_page
+    # Input is non-zero indexing, but we map to zero indexing (e.g. input 1-3 maps to 0-2) but we still return non-zero indexing
```

Comment:
Why would we want the input to be non-zero indexing? I think consumers of the endpoint (aka you, prannoy) would prefer zero indexing anyways?

Reply:
We talked about this and we thought it made more sense for the frontend to use non-zero indexing to match the page indices of the table. Like if the frontend wants data for pages 1-3 on the table, it is nicer for them to input indices 1 and 3 instead of 0 and 2.

Reply:
Okay fair enough